packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2629,4 +2631,222 @@ export class ProjectService {
             },
         );
     }
+
+    async _calculateTotal(
+        user: SessionUser,
+        projectUuid: string,
+        exploreName: string,
+        metricQuery: MetricQuery,
+    ) {
+        const totalQuery: MetricQuery = {
+            ...metricQuery,
+            limit: 1,
+            tableCalculations: [],
+            sorts: [],
+            dimensions: [],
+            metrics: metricQuery.metrics,
+            additionalMetrics: metricQuery.additionalMetrics,
+        };
+
+        const results = await this.runMetricQuery({
+            user,
+            projectUuid,
+            metricQuery: totalQuery,
+            exploreName,
+            csvLimit: undefined,
+            context: QueryExecutionContext.CALCULATE_TOTAL,
+        });
+        return results.rows[0];
+    }
+
+    async _calculateTotalFromCacheOrWarehouse(
+        user: SessionUser,
+        projectUuid: string,
+        exploreName: string,
+        metricQuery: MetricQuery,
+        invalidateCache: boolean,
+    ) {
+        return wrapOtelSpan(
+            'ProjectService._calculateTotalFromCacheOrWarehouse',
+            {},
+            async (span) => {
+                // TODO: put this hash function in a util somewhere
+                const queryHash = crypto
+                    .createHash('sha256')
+                    .update(`${projectUuid}.${JSON.stringify(metricQuery)}`)
```

Comment:
The hash should be based on the generated SQL that is going to calculate the totals. Generated SQL, to take into account the SQL from the latest explores and the user attributes. 

Instead of duplicating the code from `getResultsFromCacheOrWarehouse` I would reuse it. 

Example private method that would be used by both endpoints ( public methods )
```
private async _calculateTotalFromCacheOrWarehouse(
       user: SessionUser,
        projectUuid: string,
        exploreName: string,
        metricQuery: MetricQuery,
        invalidateCache: boolean,
) {

      // generate metric query for total calculation
      const totalQuery: MetricQuery = {
          ...metricQuery,
          limit: 1,
          tableCalculations: [],
          sorts: [],
          dimensions: [],
          metrics: metricQuery.metrics,
          additionalMetrics: metricQuery.additionalMetrics,
      };

      // get warehouse client, start ssh tunnel, etc 
      
      const { rows, cacheMetadata } = await this.getResultsFromCacheOrWarehouse({
          projectUuid,
          context, // QueryExecutionContext.CALCULATE_TOTAL
          warehouseClient,
          metricQuery,
          query,
          queryTags,
          invalidateCache,
      });
}
```